tests/common/test_cgroupstelemetry.py

```diff
@@ -392,6 +438,51 @@ def test_cgroup_is_tracked(self):
         self.assertFalse(CGroupsTelemetry.is_tracked("not_present_dummy_extension_{0}".format(i), 'cpu'))
         self.assertFalse(CGroupsTelemetry.is_tracked("not_present_dummy_extension_{0}".format(i), 'memory'))
 
+    @skip_if_predicate_false(CGroupConfigurator.get_instance().enabled, "CGroups not supported in this environment")
+    def test_telemetry_with_tracked_cgroup(self):
+        num_polls = 5
+        name = "test-cgroup"
+
+        cgs = make_self_cgroup(name)
+
+        self.assertEqual(len(cgs), 2)
+
+        for cgroup in cgs:
+            CGroupsTelemetry.track_cgroup(cgroup)
+
+        for i in range(num_polls):
```

Comment:
I don't quite get the intention of the loop...
It'd be nice if you can consume memory/cpu in parallel this loop and check that each iteration reports an increasing amount of resources. That would also allow you to make an assertion stronger than >= 0

Reply:
This loop is consuming `cpu` and `memory` after every poll. The issue with tracking the self cgroup is that it cannot be a long-running extension, and path of self is only `/sys/fs/cgroup/cpu` & `/sys/fs/cgroup/memory`. And for more exact calculation, it'll require a different test in the E2E side. I can add that task.

```
for i in range(num_polls):
    CGroupsTelemetry.poll_all_tracked()
    consume_cpu_time()  # Eat some CPU
    consume_memory()  # Eat some Memory
```